www/scripts/codecheckerviewer/filterHelper.js

```diff
@@ -211,59 +262,114 @@ function (declare, Button, ContentPane, Filter) {
      * createOverviewFilter()) to ReportFilter thrift object.
      */
-    filterToReportFilter : function (filter) {
+    filterToReportFilter : function (filter, fv) {
       var reportFilter = new CC_OBJECTS.ReportFilter();
-      var filterValues = filter.getValues();
-      
-      if (!filterValues.checker)
-        filterValues.checker = '';
 
-      reportFilter.filepath = '*' + filterValues.path + '*';
-      reportFilter.suppressed = filterValues.suppression === 'supp';
+      if (fv === undefined){
+        // Setup default values for the report filter.
+        reportFilter.filepath = '**';
+        reportFilter.suppressed = false;
+        reportFilter.checkerId = null;
+        reportFilter.severity = null;
+      }
+      else {
+        var uiValues = filter.getValues();
+
+        // Compare UI values to the provided filter values, if they are different
+        // use the UI values to filter.
+        // UI values might be empty on page load!
+        if (uiValues.suppression === fv.suppression || !uiValues.suppression) {
+          reportFilter.suppressed = fv.suppression === 'supp';
+        } else {
+          reportFilter.suppressed = uiValues.suppression === 'supp';
+        }
 
-      var index = filterValues.checker.indexOf('#');
-      var type = filterValues.checker.substr(0, index);
-      var value = filterValues.checker.substr(index + 1);
+        if (uiValues.filepath === fv.filepath) {
+          reportFilter.filepath = '*' + fv.path + '*';
+        } else {
+          reportFilter.filepath = '*' + uiValues.path + '*';
+        }
 
-      switch (type) {
-        case 'severity': reportFilter.severity = parseInt(value); break;
-        case 'checker': reportFilter.checkerId = value; break;
+        if (fv.checker === 'all'){
+          reportFilter.checkerId = null;
+        }
+        else{
+          // The filter checker value is not all it can be a severity level
+          // or a concrete checker.
+          var index = fv.checker.indexOf('#');
+          var type = fv.checker.substr(0, index);
+          var value = fv.checker.substr(index + 1);
+
+          switch (type) {
+            case 'severity': reportFilter.severity = parseInt(value); break;
+            case 'checker': reportFilter.checkerId = value; break;
+          }
+        }
       }
 
       return reportFilter;
     },
 
-    filterGroupToReportFilters : function (filterGroup) {
+    filterGroupToReportFilters : function (filterGroup, filters) {
       var that = this;
 
       var allReportFilters = [];
       filterGroup.getChildren().forEach(function (filter) {
-        allReportFilters.push(that.filterToReportFilter(filter));
+        if(filters === undefined){
+          allReportFilters.push(that.filterToReportFilter(filter, filters));
+        }
+        else{
+          var i = 0; 
+          allReportFilters.push(that.filterToReportFilter(filter, filters[i]));
```

Comment:
The variable `i` is always `0` here because it was declared as a local variable. (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var). In this case this line is same as this one:
```javascript
allReportFilters.push(that.filterToReportFilter(filter, filters[0]));
```